core/lib/queued_job_processor/src/lib.rs

```diff
@@ -138,4 +162,8 @@ pub trait JobProcessor: Sync + Send {
         started_at: Instant,
         artifacts: Self::JobArtifacts,
     ) -> anyhow::Result<()>;
+
+    fn max_attempts(&self) -> u32;
+
+    async fn get_job_attempts(&self, job_id: &Self::JobId) -> anyhow::Result<Option<u32>>;
```

Comment:
What does an error returned from this method mean? ("Number of attempts is not available at the time"?) Is it significantly different from `Ok(None)`? E.g., `wait_for_task` returns early if `get_job_attempts` fails – should it do that, given that the number of attempts is not mandatory anyway?

Reply:
This is for DB errors. `Ok(None)` is for cases when attempts are not applicable/should be ignored -- currently it's used in old prover stack + contract verifier. For old prover components I didn't want to add DAL methods given that we are going to remove these components in a few weeks. For contract verifier it's always 1 attempt.

Now I'm thinking that probably it makes sense to remove `Option` from the interface and just return `1` for old prover components, contract verifier.